pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -0,0 +1,267 @@
+// of all build configurations in the provided namespace
+type isFromBuildConfigLoader struct {
+	namespace string
+	lister    client.ImageStreamsNamespacer
+	bc        *bcLoader
+	items     []imageapi.ImageStream
+}
+
+// Load image streams found in image change triggers of
+// all build configurations in the provided namespace
+func (l *isFromBuildConfigLoader) Load() error {
+	err := l.bc.Load()
+	if err != nil {
+		return err
+	}
+	return l.loadStreams()
+}
+
+// AddToGraph adds all the loaded image streams and their tags
+// in the given graph
+func (l *isFromBuildConfigLoader) AddToGraph(g osgraph.Graph) error {
+	for i := range l.items {
+		imagegraph.EnsureImageStreamNode(g, &l.items[i])
+		imagegraph.EnsureAllImageStreamTagNodes(g, &l.items[i])
+	}
+
+	return nil
+}
+
+// loadStreams iterates over a given set of build configurations
+// and extracts all the image streams which trigger a build when
+// the image stream is updated
+func (l *isFromBuildConfigLoader) loadStreams() error {
+	avoidDuplicates := map[*imageapi.ImageStream]struct{}{}
+	for _, cfg := range l.bc.items {
+		glog.V(1).Infof("Scanning BuildConfig %q", cfg.Name)
+		for _, tr := range cfg.Spec.Triggers {
+			from := buildutil.GetImageStreamForStrategy(cfg.Spec.Strategy)
+			if tr.ImageChange != nil && from != nil && from.Name != "" {
+				glog.V(1).Infof("found ICT with from %s kind %s", from.Name, from.Kind)
+				switch from.Kind {
+				case "ImageStreamTag":
+					bits := strings.Split(from.Name, ":")
+					stream, err := l.lister.ImageStreams(l.namespace).Get(bits[0])
+					if err != nil {
+						return err
+					}
+					avoidDuplicates[stream] = struct{}{}
+				default:
+					// ImageStreamImage and DockerImage are never updated and so never
+					// trigger builds.
+					continue
+				}
+
+			}
+		}
+	}
+
+	for stream := range avoidDuplicates {
+		l.items = append(l.items, *stream)
+	}
+
+	return nil
+}
+
+// dotDump traverses the given graph starting from the provided stream node
+// and returns the stream's dependency tree in DOT syntax
+func dotDump(g osgraph.Graph, stream graph.Node, tag string) (string, error) {
+	dotGraph := dot.NewGraph()
+	attrs := make(map[string]string)
+
+	// Set graph name based on the root node
+	name := g.GraphDescriber.Name(stream)
+	dotGraph.SetName(validDOT(name))
+	// Directed graph since we illustrate dependencies
+	dotGraph.SetDir(true)
+	// Explicitly allow multiple pairs of edges between
+	// the same pair of nodes
+	dotGraph.SetStrict(false)
+	// Add root node
+	kind := g.GraphDescriber.Kind(stream)
+	attrs["label"] = fmt.Sprintf("<%s<BR /><FONT POINT-SIZE=\"10\">%s</FONT>>", name, kind)
+	dotGraph.AddNode(dotGraph.Name, g.GraphDescriber.Name(stream), attrs)
+
+	// Traverse the graph. Build the DOT graph along the way.
+	traverse(g, dotGraph, stream, tag, attrs)
+
+	// Parse DOT output for validation
+	out := dotGraph.String()
+	if _, err := dot.Parse([]byte(out)); err != nil {
```

Comment:
Nobody guarantees us that the DOT syntax we are going to put together will be correct. Reasons I can think of: careless refactoring, opinionated syntax (for example, the parser accepts only underscores from all the special characters). It's better to fail as soon as we can than producing invalid output.